src/mpid/ch4/netmod/ucx/ucx_am.c

```diff
@@ -42,16 +48,118 @@ int MPIDI_UCX_am_init(void)
     /* MPIDI_UCX_am_header_t need observe alignment */
     MPIR_Assert((sizeof(MPIDI_UCX_am_header_t) & 0x7) == 0);
 
+    for (int i = 0; i < MPIDI_UCX_AM_BUF_COUNT; i++) {
+        mpi_errno = am_post_recv(i);
```

Comment:
Have we measured the impact of this on small message latency? Thinking about this some more, the new scheme will impose a (fixed) matching penalty on all incoming pt2pt messages that do not go thru the AM path.

Reply:
I'll measure it.